api/V1/AuthenticateUser.js

```diff
@@ -57,21 +57,46 @@ module.exports = function(app) {
         request.body.password
       );
       if (passwordMatch) {
+        const token = await auth.createEntityJWT(request.body.user);
         const userData = await request.body.user.getDataValues();
-        const data = request.body.user.getJwtDataValues();
-        data._type = "user";
-        return responseUtil.send(response, {
+        responseUtil.send(response, {
           statusCode: 200,
           messages: ["Successful login."],
-          token: "JWT " + jwt.sign(data, config.server.passportSecret),
+          token: "JWT " + token,
           userData: userData
         });
       } else {
-        return responseUtil.send(response, {
+        responseUtil.send(response, {
           statusCode: 401,
           messages: ["Wrong password or username."]
         });
       }
     })
   );
+
+
+  /**
+   * @api {post} /token Generate temporary JWT
+   * @apiName Token
+   * @apiGroup Authentication
+   * @apiDescription It is sometimes necessary to include an
+   * authentication token in a URL but it is not safe to provide a
+   * user's primary JWT as it can easily leak into logs etc. This API
+   * generates a short-lived token which can be used as part of URLs.
+   *
+   * @apiUse V1UserAuthorizationHeader
+   * @apiSuccess {JSON} token JWT that may be used to call the report endpoint.
+   */
+  app.post(
+    "/token",
+    [auth.authenticateUser],
+    middleware.requestWrapper(async (request, response) => {
+      const token = auth.createEntityJWT(request.user, { expiresIn: 60 * 5 });
```

Comment:
Do we really need this to be 5 minutes? The shorter the batter I would think.

Reply:
Maybe add a "eventReport" type for signed JWTs so they are limited to just being used to get event reports. So you could have something like `auth.createEventReportJWT(user)` and that would return a JWT that can just be used for getting reports for the next minute?

Reply:
I will make it shorter and I will add limiting in a follow up PR. There's a bunch of other auth related things I want to fix up.

Reply:
OK. Sounds good.